공공데이터포털(www.data.go.kr) - 서울특별시 대기질 자료(미세먼지, 초미세먼지) 데이터

In [1]:
import pandas as pd

#### 가) 데이터 불러오기 및 열 이름 변경

##### a) 다운받은 csv 형식의 파일을 pnadas로 불러오세요. (변수명 : data_raw)

In [9]:
data_raw = pd.read_csv('서울시 대기질 자료 제공_2020-2021.csv', encoding = 'cp949')
data_raw.head(3)

일시   구분  미세먼지(PM10)  초미세먼지(PM25)
0  2021-05-31 23:00   평균        43.0         23.0
1  2021-05-31 23:00  강남구        42.0         24.0
2  2021-05-31 23:00  강동구        47.0         30.0

##### b) 데이터셋의 속성을 확인하세요. 위 데이터는 몇행 몇열로 이루어져 있나요?

In [6]:
data_raw.shape

(322606, 4)

A. 322606행, 4열

##### c) 열 이름을 아래와 같이 변경하세요.
일시 : DateTime, 구분 : Gu, 미세먼지(PM10) : PM10, 초미세먼지(PM25) : PM2.5

In [8]:
data_raw.columns = ['DateTime', 'Gu', 'PM10', 'PM2.5']
data_raw.columns

Index(['DateTime', 'Gu', 'PM10', 'PM2.5'], dtype='object')

In [12]:
data_raw.rename(columns = {'일시':'DateTime', '구분':'Gu', '미세먼지(PM10)':'PM10', '초미세먼지(PM25)':'PM2.5'}, inplace = True)
data_raw.columns

Index(['DateTime', 'Gu', 'PM10', 'PM2.5'], dtype='object')

`.columns`는 모든 column의 갯수만큼 반환해야하고, `rename`은 필요한 피쳐의 이름만 변경 가능

#### 나) 변환하기

##### a) 위에서 'DateTime' 열의 데이터 클래스는 무엇인가요? (문자 or 숫자)

In [18]:
data_raw.dtypes

DateTime     object
Gu           object
PM10        float64
PM2.5       float64
dtype: object

A. 문자 클래스

##### b) 아래 pandas의 to_datetime 함수를 이용하여 시간 타입(datetime)으로 변환하여 datetime이라는 변수를 추가하세요.